### PR TITLE
copr: Use new row format on table scan

### PR DESCRIPTION
**Pull request by niedhui**
Signed-off-by: niedhui <niedhui@gmail.com>

###  What have you changed?

This PR tries using the new row format on table-scan. The implementation is simple, grub the need columns then encode it using the old format.

**NOTE**  the code is not well-organized yet, it needs more unit test and some benchmarks, I want some suggestion/feedback so I open this PR.  Friendly ping @breeswish  @lonng 

###  What is the type of the changes?

- New feature

###  How is the PR tested?

- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

**Comment by breeswish** [comment]
/bench +sysbench +tpch tidb=pr/12634
```
workload:
  threads: 128
  prepare: true
```

File: components/tidb_query/src/batch/executors/table_scan_executor.rs [hunk 1]
```diff
@@ -142,6 +143,85 @@ struct TableScanExecutorImpl {
...
+                    .write_v2_as_datum(&row.values()[start..offset], &self.schema[*idx])?;
+                *decoded_columns += 1;
+                self.is_column_filled[*idx] = true;
+            } else if row.search_in_null_ids(*col_id) {
+                columns[*idx].mut_raw().push(datum::DATUM_DATA_NULL);
+                *decoded_columns += 1;
+                self.is_column_filled[*idx] = true;
+            } else {
+                // This column is missing. It will be filled with default values later.
+            }
```
Comment (lonng) [inline components/tidb_query/src/batch/executors/table_scan_executor.rs]: Prefer to use `continue` in `else` case and move (reuse) the following code
```rust
                *decoded_columns += 1;
                self.is_column_filled[*idx] = true;
```

**Pull request merged**